plugins/saveactions.c

```diff
@@ -304,6 +314,323 @@ static void instantsave_document_new_cb(GObject *obj, GeanyDocument *doc, gpoint
 }
 
 
+static gboolean is_temp_saved_file(const gchar *filename)
+{
+	return g_str_has_prefix(filename, PERSISTENT_TEMP_FILE_NAME_PREFIX);
+}
+
+
+static gchar* create_new_temp_file_name(void)
+{
+	GDir *dir;
+	GError *error = NULL;
+	gchar *utf8_filename, *temp_file_number_str;
+	gint temp_file_number, max_temp_file_number;
+	const gchar *filename;
+
+	dir = g_dir_open(persistent_temp_files_target_dir, 0, &error);
+	if (dir == NULL)
+	{
+		dialogs_show_msgbox(GTK_MESSAGE_ERROR, _("Persistent temp files directory not found"));
+		return NULL;
+	}
+
+	static glong temp_file_name_prefix_len = strlen(PERSISTENT_TEMP_FILE_NAME_PREFIX);
+
+	max_temp_file_number = 0;
+
+	foreach_dir(filename, dir)
+	{
+		utf8_filename = utils_get_utf8_from_locale(filename);
```

Comment:
The conversion is probably an overkill here - the file name will just always be ASCII so `locale == UTF8`.

Reply:
removed

Reply:
removed